src/Core/Framework/Plugin/PluginLifecycleService.php

```diff
@@ -508,4 +528,25 @@ private function getPluginInstance(string $pluginBaseClassString): Plugin
 
         return $this->getPluginBaseClass($pluginBaseClassString);
     }
+
+    private function savePluginConfiguration(Plugin $plugin, InstallContext $context): void
+    {
+        $config     = $this->configReader->getConfigFromBundle($plugin);
+        $prefix     = $plugin->getName() . '.config.';
+        $noOverride = $context instanceof UpdateContext;
+
+        foreach ($config as $card) {
+            foreach ($card['elements'] as $element) {
+                $key   = $prefix . $element['name'];
+                $value = $element['defaultValue'];
```

Comment:
here we should cast the value to the type of the option

Reply:
I would cast the values like the following. Do you agree?

- int, single-select => int
- bool => bool
- float => float

Everything else (picked from [here](https://docs.shopware.com/en/shopware-platform-dev-en/internals/plugins/plugin-config#the-different-types-of-input-field)) should be a string.

Reply:
not sure about the single-select. this could also be a string. 
everything else is ok :+1:

Reply:
I have applied this change, too.